x-pack/plugins/triggers_actions_ui/public/application/lib/action_variables.ts

```diff
@@ -299,136 +166,194 @@ function getAlwaysProvidedActionVariables(): ActionVariable[] {
     description: i18n.translate('xpack.triggersActionsUI.actionVariables.newAlertsCountLabel', {
       defaultMessage: 'The count of new alerts.',
     }),
-  });
-  result.push({
-    name: SummaryAlertProvidedActionVariables.newAlertsData,
+  },
+  [SummaryAlertProvidedActionVariables.newAlertsData]: {
     description: i18n.translate('xpack.triggersActionsUI.actionVariables.newAlertsDataLabel', {
       defaultMessage: 'An array of objects for new alerts.',
     }),
-  });
-  result.push({
-    name: SummaryAlertProvidedActionVariables.ongoingAlertsCount,
+  },
+  [SummaryAlertProvidedActionVariables.ongoingAlertsCount]: {
     description: i18n.translate('xpack.triggersActionsUI.actionVariables.ongoingAlertsCountLabel', {
       defaultMessage: 'The count of ongoing alerts.',
     }),
-  });
-  result.push({
-    name: SummaryAlertProvidedActionVariables.ongoingAlertsData,
+  },
+  [SummaryAlertProvidedActionVariables.ongoingAlertsData]: {
     description: i18n.translate('xpack.triggersActionsUI.actionVariables.ongoingAlertsDataLabel', {
       defaultMessage: 'An array of objects for ongoing alerts.',
     }),
-  });
-  result.push({
-    name: SummaryAlertProvidedActionVariables.recoveredAlertsCount,
+  },
+  [SummaryAlertProvidedActionVariables.recoveredAlertsCount]: {
     description: i18n.translate(
       'xpack.triggersActionsUI.actionVariables.recoveredAlertsCountLabel',
       {
         defaultMessage: 'The count of recovered alerts.',
       }
     ),
-  });
-  result.push({
-    name: SummaryAlertProvidedActionVariables.recoveredAlertsData,
+  },
+  [SummaryAlertProvidedActionVariables.recoveredAlertsData]: {
     description: i18n.translate(
       'xpack.triggersActionsUI.actionVariables.recoveredAlertsDataLabel',
       {
         defaultMessage: 'An array of objects for recovered alerts.',
       }
     ),
-  });
-  result.push({
-    name: SummaryAlertProvidedActionVariables.allAlertsCount,
-    description: i18n.translate('xpack.triggersActionsUI.actionVariables.allAlertsCountLabel', {
-      defaultMessage: 'The count of all alerts.',
-    }),
-  });
-  result.push({
-    name: SummaryAlertProvidedActionVariables.allAlertsData,
-    description: i18n.translate('xpack.triggersActionsUI.actionVariables.allAlertsDataLabel', {
-      defaultMessage: 'An array of objects for all alerts.',
-    }),
-  });
+  },
+};
 
-  return result;
+function prefixKeys(actionVariables: ActionVariable[], prefix: string): ActionVariable[] {
+  return actionVariables.map((actionVariable) => {
+    return { ...actionVariable, name: `${prefix}${actionVariable.name}` };
+  });
 }
+
+const transformContextVariables = (
+  variables: Record<string, ActionVariablesWithoutName>
+): ActionVariable[] =>
+  Object.entries(variables).map(([key, variable]) => ({ ...variable, name: key }));
+
+export const getAlwaysProvidedActionVariables = (): ActionVariable[] => {
+  const variables = cloneDeep(AlertProvidedActionVariableDescriptions);
```

Comment:
do we actually need to deep clone if line 347 is already doing a shallow clone?

Reply:
I am not sure how this function is being used and will be used in the future. I would like to avoid someone messing up with this static object. Now it is called in line 347 but who knows how it will be used in the future.

Reply:
maybe you can freeze it? So if people want to modify it they can just copy it themselves

Reply:
Freezing is a good idea. I will do that.